python_modules/dagster/dagster/_core/definitions/asset_daemon_context.py

```diff
@@ -485,37 +485,26 @@ def build_run_requests_with_backfill_policies(
             # non partitioned assets will be backfilled in a single run
             run_requests.append(RunRequest(asset_selection=list(asset_keys), tags=tags))
         else:
-            backfill_policies = {
-                check.not_none(asset_graph.get(asset_key).backfill_policy)
-                for asset_key in asset_keys
-            }
-            if len(backfill_policies) == 1:
-                # if all backfill policies are the same, we can backfill them together
-                backfill_policy = backfill_policies.pop()
+            asset_keys_by_backfill_policy = defaultdict(set)
+            for asset_key in asset_keys:
+                asset_keys_by_backfill_policy[asset_graph.get(asset_key).backfill_policy].add(
+                    asset_key
+                )
+
+            for backfill_policy, aks in asset_keys_by_backfill_policy.items():
```

Comment:
Would it make sense to do all the grouping in one place? I.e. `assets_to_reconcile_by_partitions_def_partition_keys_backfill_policy`?

Reply:
yeah good point - done!